### PR TITLE
Bump substrate to polkadot-v0.9.38

### PR DESCRIPTION
**Pull request by dmitrylavrenov**
## Frontier changes

https://github.com/humanode-network/frontier/compare/locked/polkadot-v0.9.36...humanode-network:frontier:master

### Most interesting commits

- [x] [feat: contract deployment callback](https://github.com/humanode-network/frontier/commit/ede6023b6a0b0d1dbb1952fec3dd9027fa881c8c)
- [x] [improve weight per gas calculation](https://github.com/humanode-network/frontier/commit/40083d8f42fbec8185eb4cfeedefa3a4bed3c2c5)
- [x] [feat: only use substrate-wasm-builder when std feature is enabled](https://github.com/humanode-network/frontier/commit/898580c86c525bc428133eff66f0ed551ccc757e)
- [x] [Move the weight_per_gas function into fp-evm](https://github.com/humanode-network/frontier/commit/7c444e3eb1731b78e3470dfc2d0f6738142e5e09)
- [x] [feat: add fc-storage](https://github.com/humanode-network/frontier/commit/b984ff19acd20a3434f2b2ff481266413c9b8f4f)
- [x] [feat: add a new post log type into header digest and provide a compatible mapping-sync worker](https://github.com/humanode-network/frontier/commit/0535842137360e80f36bbe3d6a5333d6ad8b9c8a)
- [x] [feat: provide a parameter for the runtime config to choose what to include in the PostLog](https://github.com/humanode-network/frontier/commit/98aacf69e976340338f81455e48e8b330007e544)
- [x] [Add EGA for all impl blocks taking Eth struct](https://github.com/humanode-network/frontier/commit/4afc9d413c7d9dd8a7045d48e72691f38ab769dc)

## Substrate changes

https://github.com/humanode-network/substrate/compare/polkadot-v0.9.36...humanode-network:substrate:polkadot-v0.9.38

### Most interesting commits

- [x] [frame_support::storage: Add StorageStreamIter](https://github.com/humanode-network/substrate/commit/a96cb5725407f8cc5ae7c9efc85a80548d071833)
- [x] [Support custom genesis block](https://github.com/humanode-network/substrate/commit/cad470b6b68d6d2c79371dd67d2f915b37ab8304)
- [x] [rpc: Implement chainHead RPC API](https://github.com/humanode-network/substrate/commit/e118df551952d1aa7d85e79f04e1b8cb851b6761)
- [x] [NFTs 2.0](https://github.com/humanode-network/substrate/commit/c7f5df2c160e8d836285127a60bbf7793b7067ba)
- [x] [Ignore empty authority changes](https://github.com/humanode-network/substrate/commit/ed31016ca2bfe24438c17e81fc98b97738af55ff)
- [x] [babe: allow skipping over empty epochs](https://github.com/humanode-network/substrate/commit/999b1a18b084d9d87cd0cc0f230228e00fa15be6)
- [x] [Make CLI state pruning optional again](https://github.com/humanode-network/substrate/commit/27e9a1c0fdb200c3736f513f3292b8b954e3d021)
- [x] [Improve inactive fund tracking](https://github.com/humanode-network/substrate/commit/d2ef4b061f3802d7aa4254d6ce1f15b261a0fa61)
- [x] [Print "stalled" task on shutdown](https://github.com/humanode-network/substrate/commit/891d6a5c870ab88521183facafc811a203bb6541)
- [x] [Add ensure-ops family methods](https://github.com/humanode-network/substrate/commit/043a60b0f25c7b2abfcb213fa0af6fef946198c2)
- [x] [try-runtime: add cli option --export-proof](https://github.com/humanode-network/substrate/commit/55c64bcc2af5a6e5fc3eb245e638379ebe18a58d)
- [x] [Selectable on-runtime-upgrade checks](https://github.com/humanode-network/substrate/commit/a052224682994e6b54c94c17572e061f690149e5)
- [x] [grandpa: pass the actual best block to voting rules](https://github.com/humanode-network/substrate/commit/da3a34ff7861d9bb504d013f2ef89e8b0ce2017c)
- [x] [Only use substrate-wasm-builder when std feature is enabled](https://github.com/humanode-network/substrate/commit/90ffe716b88f6dc64eeb530332267a86e66668b4)
- [x] [Stop keeping track of epoch changes for the sync gap](https://github.com/humanode-network/substrate/commit/2bfc1dd66ef32cf8beb90007dfb544a9d28f1b2f)
- [x] [Use the balance trait as we have one](https://github.com/humanode-network/substrate/commit/cbc525f271146ba9c684736bfc5c26f77e849dae)
- [x] [txpool: don't maintain the pool during major sync](https://github.com/humanode-network/substrate/commit/7ff6fa2bfa47f288216fdc6e1585f69756c5ec75)
- [x] [[client/network] Add support for /wss addresses](https://github.com/humanode-network/substrate/commit/28937302694787cd1e6dbd846ff2ba4b1cf82025)
- [x] [frame-benchmarking: Macros should not force a particular env](https://github.com/humanode-network/substrate/commit/42f38dbd961342cd965154858e50f0253b0b7c0a)
- [x] [Notification-based block pinning](https://github.com/humanode-network/substrate/commit/d4384f242f48fd62516aac02c3f18b5a8c9e5793)
- [x] [Make DispatchError impl MEL](https://github.com/humanode-network/substrate/commit/0b6aec52a90870c999856cd37f7d04789cdd8dfc)
- [x] [Warn validators with slow hardware](https://github.com/humanode-network/substrate/commit/38ec2d8957d843fa424397f79a484d773eea1454)
- [x] [Move slow hardware warning print logic to CLI](https://github.com/humanode-network/substrate/commit/355e4a9c87f85637185c0852bda9aeb3db404b6d)
- [x] [service: storage monitor added](https://github.com/humanode-network/substrate/commit/ca5a505b7085f7af51eb76c686cc80b55bf6d5ed)
- [x] [Babe: bad epoch index with skipped epochs and warp sync](https://github.com/humanode-network/substrate/commit/e3938747cdc201a6c3fbf920d0e983e2b7be01c7)
- [x] [Add WeightToFee and LengthToFee impls to transaction-payment Runtime API](https://github.com/humanode-network/substrate/commit/2dbf62591fd35725c6e39f8ddfd27c3695dab594)
- [x] [Add Proof Size to Weight Output](https://github.com/humanode-network/substrate/commit/ff9921a260a67e3a71f25c8b402cd5c7da787a96)
- [x] [Remove uncles related code](https://github.com/humanode-network/substrate/commit/fdff2b23009f25dd42d3a12fee6c998864399de3)
- [x] [grandpa: cleanup stale entries in set id session mapping](https://github.com/humanode-network/substrate/commit/129fee774a6d185d117a57fd1e81b3d0d05ad747)
- [x] [Fix block pruning](https://github.com/humanode-network/substrate/commit/5a281ced23f8802b44d2a0fb6e1e08f6013b2554)
- [x] [grandpa: don't error if best block and finality target are inconsistent](https://github.com/humanode-network/substrate/commit/15e218a5c777b5e19fa7106be88f87852c3ac4a3)

### CODE REVIEW

**Comment by MOZGIII** [comment]
https://github.com/paritytech/substrate/pull/13289 is also relevant

**Comment by dmitrylavrenov** [comment]
@MOZGIII checked practically that [Babe: bad epoch index with skipped epochs and warp sync](https://github.com/humanode-network/substrate/commit/e3938747cdc201a6c3fbf920d0e983e2b7be01c7) fixes slot value and prevent having an error with `Unexpected epoch change`.

File: crates/humanode-peer/src/benchmarking.rs [hunk 1]
```diff
@@ -97,11 +97,6 @@ pub fn inherent_benchmark_data(config: &Configuration) -> sc_cli::Result<Inheren
     futures::executor::block_on(timestamp.provide_inherent_data(&mut inherent_data))
         .map_err(|e| format!("creating timestamp inherent data: {:?}", e))?;
 
-    let uncles =
-        sp_authorship::InherentDataProvider::<<Block as BlockT>::Header>::check_inherents();
```
Comment (MOZGIII) [inline crates/humanode-peer/src/benchmarking.rs:101]: Looks like this causes the
```
unused import: `opaque::Block`
```
 above

**Review by MOZGIII — APPROVED** [review]

File: crates/humanode-peer/src/benchmarking.rs (hunk 1 above)
Comment (MOZGIII) [reply crates/humanode-peer/src/benchmarking.rs:101]: I think I fixed it with the latest commit fbe63ace

**Pull request merged**